File: NBE_5_7_2_02TEAM/src/main/java/io/twogether/nbe_5_7_2_02team/tag/service/TagService.java
package io.twogether.nbe_5_7_2_02team.tag.service;

import io.twogether.nbe_5_7_2_02team.tag.dao.TagRepository;
import io.twogether.nbe_5_7_2_02team.tag.domain.Tag;

import lombok.RequiredArgsConstructor;

import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;

@Service
@RequiredArgsConstructor
public class TagService {

    private final TagRepository tagRepository;

    @Transactional(readOnly = true)
    public List<Tag> getAllTags() {
        return tagRepository.findAll();
    }
}

File: NBE_5_7_2_02TEAM/src/main/java/io/twogether/nbe_5_7_2_02team/tag/service/TagService.kt
package io.twogether.nbe_5_7_2_02team.tag.service

import io.twogether.nbe_5_7_2_02team.tag.dao.TagRepository
import io.twogether.nbe_5_7_2_02team.tag.domain.Tag
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
@Transactional(readOnly = true)
class TagService(
    private val tagRepository: TagRepository
) {

    fun getAllTags(): List<Tag> = tagRepository.findAll()
}
